tests/test_modules.py

```diff
@@ -776,3 +805,36 @@ def forward(self, x):
         self.assertEqual(cnt.op_count, 1)
         self.assertTrue(torchdynamo.testing.same(pre, opt_pre))
         self.assertTrue(torchdynamo.testing.same(out1, out_post))
+
+    def test_lazy_module(self):
+        input_shape = (16, 3, 6, 7, 8)
+        cnt = torchdynamo.testing.CompileCounter()
+        with torchdynamo.optimize(cnt):
+
+            def test_dynamic_module():
+                module = LazyModule()
```

Comment:
Construct this outside of torchdynamo.optimize(), otherwise it gets mapped to UnspecializedNNModule()

Reply:
Do I want to test that case too? I also have a check in UnspecializedNNModule to handle lazy modules.

Reply:
Yes we should check them both.

Reply:
Should be good now